src/agent/mdns_mojo.cpp

```diff
@@ -0,0 +1,237 @@
+    MOJO_CONNECTOR_NS::ExternalConnector::Connect(
+        chromecast::external_mojo::GetBrokerPath(),
+        base::BindOnce(&MdnsMojoPublisher::mMojoConnectCb, base::Unretained(this)));
+}
+
+void MdnsMojoPublisher::mMojoConnectCb(std::unique_ptr<MOJO_CONNECTOR_NS::ExternalConnector> aConnector)
+{
+    if (aConnector)
+    {
+        otbrLog(OTBR_LOG_INFO, "Mojo connected");
+        aConnector->set_connection_error_callback(
+            base::BindOnce(&MdnsMojoPublisher::mMojoDisconnectedCb, base::Unretained(this)));
+        aConnector->BindInterface("chromecast", &mResponder);
+        mConnector = std::move(aConnector);
+        Start();
+    }
+    else
+    {
+        mMojoTaskRunner->PostTask(FROM_HERE, base::BindOnce(&MdnsMojoPublisher::ConnectToMojo, base::Unretained(this)));
+    }
+}
+
+void MdnsMojoPublisher::mMojoDisconnectedCb()
+{
+    mConnector = nullptr;
+}
+
+otbrError MdnsMojoPublisher::Start()
+{
+    otbrError err = OTBR_ERROR_NONE;
+
+    VerifyOrExit(mConnector != nullptr, err = OTBR_ERROR_MDNS);
+
+    mStarted = true;
+    mStateHandler(mContext, kStateReady);
+exit:
+    return err;
+}
+
+bool MdnsMojoPublisher::IsStarted() const
+{
+    return mStarted;
+}
+
+void MdnsMojoPublisher::Stop()
+{
+    mMojoTaskRunner->PostTask(FROM_HERE, base::BindOnce(&MdnsMojoPublisher::StopPublishTask, base::Unretained(this)));
+}
+
+void MdnsMojoPublisher::StopPublishTask(void)
+{
+    if (!mLastServiceName.empty())
+    {
+        mResponder->UnregisterServiceInstance(mLastServiceName, mLastInstanceName, base::DoNothing());
+    }
+    mLastServiceName.clear();
+    mLastInstanceName.clear();
+}
+
+otbrError MdnsMojoPublisher::PublishService(uint16_t aPort, const char *aName, const char *aType, ...)
+{
+    otbrError                err = OTBR_ERROR_NONE;
+    std::vector<std::string> text;
+    va_list                  args;
+
+    va_start(args, aType);
+
+    VerifyOrExit(mConnector != nullptr, err = OTBR_ERROR_MDNS);
+    for (const char *name = va_arg(args, const char *); name; name = va_arg(args, const char *))
+    {
+        const char *value = va_arg(args, const char *);
+
+        text.emplace_back(std::string(name) + "=" + std::string(value));
+    }
+    mMojoTaskRunner->PostTask(FROM_HERE, base::BindOnce(&MdnsMojoPublisher::PublishServiceTask, base::Unretained(this),
+                                                        aPort, aType, aName, text));
+
+exit:
+    va_end(args);
+    return err;
+}
+
+void MdnsMojoPublisher::PublishServiceTask(uint16_t                        aPort,
+                                           const std::string &             aType,
+                                           const std::string &             aInstanceName,
+                                           const std::vector<std::string> &aText)
+{
+    std::string            serviceName;
+    std::string            serviceProtocol;
+    std::string::size_type split = aType.rfind('.');
```

Comment:
is it possible to simplify to:
```
std::string::size_type split = aType.rfind('.', aType.length() - 2);
```
line 156-159 can be saved.

Reply:
Required to handle aType=="a" or aType=="." corner cases